openmc/source.py

```diff
@@ -329,6 +329,41 @@ def from_xml_element(cls, elem: ET.Element) -> 'openmc.Source':
 
         return source
 
+    @classmethod
+    def from_cell_with_material(cls, cell):
+        """Generate an isotropic photon source from an openmc.Cell object.
+
+        Parameters
+        ----------
+        cell : openmc.Cell
+            OpenMC cell object to use for the source creation
+
+        Returns
+        -------
+        openmc.Source
+            Source generated from an openmc.Material
+
+        """
+
+        if cell.fill is None:
+            msg = ("The cell must be filled with an openmc.Material object "
+                   "as this method make use of the cells material and the " 
+                   "Material.decay_photon_energy method.")
+            raise ValueError(msg)
+        if cell.fill.volume is None:
+            msg = ("The openmc.Material object that the cell is filled with "
+                   "must have the volume property set")
+            raise ValueError(msg)
+
+        source = cls()
+        source.domain=cell
```

Comment:
`source.domain=cell` doesn't actually correctly set the cell for rejection sampling. This has to be set in the constructor when the `Source` object is created. Perhaps this should be changed?

Reply:
Ah sorry about that mistake, thanks for spotting it Ethan